packages/sql-d1/src/D1Client.ts

```diff
@@ -0,0 +1,205 @@
+}
+
+/**
+ * @category constructor
+ * @since 1.0.0
+ */
+export const make = (
+  options: D1ClientConfig
+): Effect.Effect<D1Client, never, Scope.Scope> =>
+  Effect.gen(function*() {
+    const compiler = Statement.makeCompilerSqlite(options.transformQueryNames)
+    const transformRows = Statement.defaultTransforms(
+      options.transformResultNames!
+    ).array
+
+    const makeConnection = Effect.gen(function*() {
+      const db = options.db
+
+      const prepareCache = yield* Cache.make({
+        capacity: options.prepareCacheSize ?? 200,
+        timeToLive: options.prepareCacheTTL ?? Duration.minutes(10),
+        lookup: (sql: string) =>
+          Effect.try({
+            try: () => db.prepare(sql),
+            catch: (error) => new SqlError({ error })
+          })
+      })
+
+      const runStatement = (
+        statement: D1PreparedStatement,
+        params: ReadonlyArray<Statement.Primitive> = []
+      ) =>
+        Effect.tryPromise({
+          try: async () => {
+            const response = await statement.bind(...params).all()
+            if (response.error) {
+              throw response.error
+            }
+            return response.results || []
+          },
+          catch: (error) => new SqlError({ error })
+        })
+
+      const run = (
+        sql: string,
+        params: ReadonlyArray<Statement.Primitive> = []
+      ) => Effect.flatMap(prepareCache.get(sql), (s) => runStatement(s, params))
+
+      const runRaw = (
+        sql: string,
+        params: ReadonlyArray<Statement.Primitive> = []
+      ) => Effect.map(runStatement(db.prepare(sql), params), transformRows)
+
+      const runTransform = options.transformResultNames
+        ? (sql: string, params?: ReadonlyArray<Statement.Primitive>) => Effect.map(run(sql, params), transformRows)
+        : run
+
+      const runValues = (
+        sql: string,
+        params: ReadonlyArray<Statement.Primitive>
+      ) =>
+        Effect.flatMap(
+          prepareCache.get(sql),
+          (statement) =>
+            Effect.tryPromise({
+              try: () => {
+                return statement.bind(...params).raw() as Promise<
+                  ReadonlyArray<
+                    ReadonlyArray<Statement.Primitive>
+                  >
+                >
+              },
+              catch: (error) => new SqlError({ error })
+            })
+        )
+
+      return identity<Connection>({
+        execute(sql, params) {
+          return runTransform(sql, params)
+        },
+        executeValues(sql, params) {
+          return runValues(sql, params)
+        },
+        executeWithoutTransform(sql, params) {
+          return run(sql, params)
+        },
+        executeRaw(sql, params) {
+          return runRaw(sql, params)
+        },
+        executeStream(_sql, _params) {
+          return Effect.dieMessage("executeStream not implemented")
+        }
+      })
+    })
+
+    const semaphore = yield* Effect.makeSemaphore(1)
```

Comment:
You can remove the semaphore, as there is no transactions.

Reply:
Fixed